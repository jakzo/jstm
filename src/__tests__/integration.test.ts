import path from 'path';

import fse from 'fs-extra';

import { initProject } from '..';

interface FileTree {
  [filename: string]: string | FileTree;
}

const readFilesIntoObject = async (dir: string): Promise<FileTree> =>
  Object.assign(
    {},
    ...(await Promise.all(
      (await fse.readdir(dir, { withFileTypes: true })).map(async entry => {
        const entryPath = path.join(dir, entry.name);
        return {
          [entry.name]: await (entry.isDirectory()
            ? readFilesIntoObject(entryPath)
            : fse.readFile(entryPath, 'utf8')),
        };
      }),
    )),
  ) as FileTree;

const integrationFixtureDir = path.join(__dirname, '__fixtures__', 'integration');

test('basic', async () => {
  const projectPath = path.join(integrationFixtureDir, 'basic');
  await fse.emptyDir(projectPath);
  await fse.writeJson(path.join(projectPath, 'package.json'), {});

  await initProject(projectPath, true);

  const output = await readFilesIntoObject(projectPath);
  expect(output).toMatchSnapshot();
});

test('no package.json', async () => {
  const projectPath = path.join(integrationFixtureDir, 'no-package-json');
  await fse.emptyDir(projectPath);

  await expect(async () => initProject(projectPath, true)).rejects.toBeTruthy();
});
