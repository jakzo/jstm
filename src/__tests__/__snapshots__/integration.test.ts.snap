// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
Object {
  ".eslintignore": "# DO NOT MODIFY
# This file is generated based on .gitignore -- modify that file instead then run \`yarn install\`

# === Build Files ===
# Files which are the result of transforming other files in the project
# First item should be the path to the Typescript build output directory without any glob patterns
dist

# === Cache Files ===
# Generated by tools for caching
node_modules
.*cache
.jest
.yarn
*.tsbuildinfo
yarn.lock
package-lock.json
!yarn.lock

# === Log Files ===
# Log output files intended to be read by people rather than the project
*.log
coverage

# === External Files ===
# Files commonly created by the OS or tools not defined in this project
.DS_Store

# === Generated Files ===
# Files generated by tools from the project
/.gitignore
/.npmignore
/.prettierrc.js
/.prettierignore
/.eslintrc.js
/.eslintignore
/jest.config.js
/tsconfig.json
/tsconfig.build.json
/.nvmrc
/.vscode
!/.gitignore

# === Other Files ===
# Any other ignored files
",
  ".eslintrc.js": "module.exports = require('@jakzo/project/dist/config/eslint').config;
",
  ".gitignore": "# DO NOT CHANGE THE STRUCTURE OF THIS FILE
# You may add extra patterns but do not change the \\"=== HEADING ===\\" comments or provided patterns

# === Build Files ===
# Files which are the result of transforming other files in the project
# First item should be the path to the Typescript build output directory without any glob patterns
dist

# === Cache Files ===
# Generated by tools for caching
node_modules
.*cache
.jest
.yarn
*.tsbuildinfo
yarn.lock
package-lock.json
!yarn.lock

# === Log Files ===
# Log output files intended to be read by people rather than the project
*.log
coverage

# === External Files ===
# Files commonly created by the OS or tools not defined in this project
.DS_Store

# === Generated Files ===
# Files generated by tools from the project
/.gitignore
/.npmignore
/.prettierrc.js
/.prettierignore
/.eslintrc.js
/.eslintignore
/jest.config.js
/tsconfig.json
/tsconfig.build.json
/.nvmrc
/.vscode
!/.gitignore

# === Other Files ===
# Any other ignored files
",
  ".npmignore": "# Ignore all files and make exceptions only for required files
**

# Include package files
!/package.json
!/README.md

# Include build files
!dist

# Include source files (so that source maps work)
!/src

# Remove test files from source
!__*__
",
  ".nvmrc": "14
",
  ".prettierignore": "# DO NOT MODIFY
# This file is generated based on .gitignore -- modify that file instead then run \`yarn install\`

# === Build Files ===
# Files which are the result of transforming other files in the project
# First item should be the path to the Typescript build output directory without any glob patterns
dist

# === Cache Files ===
# Generated by tools for caching
node_modules
.*cache
.jest
.yarn
*.tsbuildinfo
yarn.lock
package-lock.json
!yarn.lock

# === Log Files ===
# Log output files intended to be read by people rather than the project
*.log
coverage

# === External Files ===
# Files commonly created by the OS or tools not defined in this project
.DS_Store

# === Generated Files ===
# Files generated by tools from the project
/.gitignore
/.npmignore
/.prettierrc.js
/.prettierignore
/.eslintrc.js
/.eslintignore
/jest.config.js
/tsconfig.json
/tsconfig.build.json
/.nvmrc
/.vscode
!/.gitignore

# === Other Files ===
# Any other ignored files
",
  ".prettierrc.js": "module.exports = require('@jakzo/project/dist/config/prettier').config;
",
  ".vscode": Object {
    "extensions.json": "{
  \\"recommendations\\": [\\"dbaeumer.vscode-eslint\\", \\"esbenp.prettier-vscode\\"]
}
",
    "settings.json": "{
  \\"[typescript]\\": {
    \\"editor.formatOnSave\\": false,
    \\"editor.codeActionsOnSave\\": {
      \\"source.fixAll\\": true
    }
  },
  \\"[typescriptreact]\\": {
    \\"editor.formatOnSave\\": false,
    \\"editor.codeActionsOnSave\\": {
      \\"source.fixAll\\": true
    }
  },
  \\"[javascript]\\": {
    \\"editor.formatOnSave\\": false,
    \\"editor.codeActionsOnSave\\": {
      \\"source.fixAll\\": true
    }
  },
  \\"[javascriptreact]\\": {
    \\"editor.formatOnSave\\": false,
    \\"editor.codeActionsOnSave\\": {
      \\"source.fixAll\\": true
    }
  },

  \\"editor.detectIndentation\\": true,
  \\"editor.tabSize\\": 2,
  \\"editor.formatOnSave\\": true,
  \\"files.eol\\": \\"\\\\n\\",
  \\"typescript.tsdk\\": \\"node_modules\\\\\\\\typescript\\\\\\\\lib\\"
}
",
  },
  "jest.config.js": "module.exports = require('@jakzo/project/dist/config/jest').config;
",
  "package.json": "{
  \\"scripts\\": {
    \\"=== Generated Scripts (do not modify) ===\\": \\"\\",
    \\"build\\": \\"tsc -p ./tsconfig.build.json\\",
    \\"build:watch\\": \\"yarn build -w\\",
    \\"lint\\": \\"yarn lint:eslint && yarn lint:prettier\\",
    \\"lint:fix\\": \\"yarn lint:eslint --fix && yarn lint:prettier --write\\",
    \\"lint:eslint\\": \\"eslint --ext js,jsx,ts,tsx ./\\",
    \\"lint:prettier\\": \\"prettier -c \\\\\\"./**/*{.json,.md}\\\\\\"\\",
    \\"test\\": \\"jest\\",
    \\"test:watch\\": \\"yarn test --watch\\",
    \\"=== (end generated scripts) ===\\": \\"\\",
    \\"\\": \\"\\"
  }
}
",
  "tsconfig.build.json": "// DO NOT MODIFY
// This file is auto-generated
{
  \\"extends\\": \\"./tsconfig.json\\",
  \\"include\\": [\\"./src\\", \\"./src/**/*\\", \\"./src/**/*.json\\"],
  \\"exclude\\": [\\"**/__*__\\", \\"**/__*__/**/*\\"],
  \\"compilerOptions\\": {
    \\"rootDir\\": \\"./src\\",
    \\"outDir\\": \\"./dist\\",
    \\"types\\": [\\"node\\"]
  }
}
",
  "tsconfig.json": "// DO NOT MODIFY
// This file is auto-generated
{
  \\"include\\": [\\"./\\", \\"./**/*.json\\", \\"**/.*\\"],
  \\"exclude\\": [
    \\"**/dist\\",
    \\"**/dist/**/*\\",
    \\"**/node_modules\\",
    \\"**/node_modules/**/*\\",
    \\"**/.*cache\\",
    \\"**/.*cache/**/*\\",
    \\"**/.jest\\",
    \\"**/.jest/**/*\\",
    \\"**/.yarn\\",
    \\"**/.yarn/**/*\\",
    \\"**/*.tsbuildinfo\\",
    \\"**/*.tsbuildinfo/**/*\\",
    \\"**/yarn.lock\\",
    \\"**/yarn.lock/**/*\\",
    \\"**/package-lock.json\\",
    \\"**/package-lock.json/**/*\\",
    \\"**/*.log\\",
    \\"**/*.log/**/*\\",
    \\"**/coverage\\",
    \\"**/coverage/**/*\\",
    \\"**/.DS_Store\\",
    \\"**/.DS_Store/**/*\\",
    \\"./.gitignore\\",
    \\"./.gitignore/**/*\\",
    \\"./.npmignore\\",
    \\"./.npmignore/**/*\\",
    \\"./.prettierrc.js\\",
    \\"./.prettierrc.js/**/*\\",
    \\"./.prettierignore\\",
    \\"./.prettierignore/**/*\\",
    \\"./.eslintrc.js\\",
    \\"./.eslintrc.js/**/*\\",
    \\"./.eslintignore\\",
    \\"./.eslintignore/**/*\\",
    \\"./jest.config.js\\",
    \\"./jest.config.js/**/*\\",
    \\"./tsconfig.json\\",
    \\"./tsconfig.json/**/*\\",
    \\"./tsconfig.build.json\\",
    \\"./tsconfig.build.json/**/*\\",
    \\"./.nvmrc\\",
    \\"./.nvmrc/**/*\\",
    \\"./.vscode\\",
    \\"./.vscode/**/*\\"
  ],
  \\"compilerOptions\\": {
    \\"outDir\\": \\"./dist/\\",
    \\"rootDir\\": \\"./\\",
    \\"baseUrl\\": \\"./\\",
    \\"target\\": \\"es2018\\",
    \\"module\\": \\"commonjs\\",
    \\"lib\\": [\\"es2018\\"],
    \\"types\\": [\\"node\\", \\"jest\\"],
    \\"jsx\\": \\"react\\",
    \\"allowJs\\": true,
    \\"resolveJsonModule\\": true,
    \\"composite\\": true,
    \\"skipLibCheck\\": true,
    \\"sourceMap\\": true,
    \\"declaration\\": true,
    \\"importHelpers\\": true,
    \\"removeComments\\": false,
    \\"stripInternal\\": false,
    \\"moduleResolution\\": \\"node\\",
    \\"allowSyntheticDefaultImports\\": true,
    \\"esModuleInterop\\": true,
    \\"emitDecoratorMetadata\\": true,
    \\"experimentalDecorators\\": true,
    \\"strict\\": true,
    \\"noImplicitAny\\": true,
    \\"allowUnreachableCode\\": true,
    \\"allowUnusedLabels\\": false,
    \\"alwaysStrict\\": true,
    \\"forceConsistentCasingInFileNames\\": true,
    \\"noFallthroughCasesInSwitch\\": true,
    \\"noImplicitReturns\\": true,
    \\"noImplicitThis\\": true,
    \\"noUnusedLocals\\": true,
    \\"noUnusedParameters\\": false,
    \\"strictFunctionTypes\\": false,
    \\"strictNullChecks\\": true,
    \\"strictPropertyInitialization\\": false
  }
}
",
}
`;
